sdk/appconfiguration/app-configuration/src/appConfigurationClient.ts

```diff
@@ -125,14 +125,24 @@ export class AppConfigurationClient {
    * @param key The name of the key.
    * @param options Optional parameters for the request.
    */
-  getConfigurationSetting(
+  async getConfigurationSetting(
     key: string,
     options: GetConfigurationSettingOptions = {}
   ): Promise<GetConfigurationSettingResponse> {
-    return this.client.getKeyValue(key, {
+    const response = await this.client.getKeyValue(key, {
       label: options.label,
-      ...options
-    });
+      ...options,
+      ifMatch: quoteETag(options.ifMatch),
+      ifNoneMatch: quoteETag(options.ifNoneMatch),
+    }) as GetConfigurationSettingResponse;
+
+    // 304 only comes back if the user has passed a conditional option in their
```

Comment:
Interesting, so a 304 status isn't considered an error when calling `client.getKeyValue`?

Reply:
There are some exceptions to the rule inside of the swagger file and this one was one that I didn't change (so it ends up manifesting as jut a returned value).

It is kind of odd but in this case it's somewhat appropriate. In reality I think the status code might more appropriately be 204, not 304.